### PR TITLE
Bad row partition

### PR DESCRIPTION
**Pull request by aldemirenes**

### CODE REVIEW

File: src/main/scala/com.snowplowanalytics.s3/loader/KinesisS3Emitter.scala [hunk 1]
```diff
@@ -108,3 +126,65 @@ class KinesisS3Emitter(
...
+    * to their schema key. Put records which are not self describing data
+    * to under "old bad row type".
+    */
+  private[loader] def partitionWithSchemaKey(records: List[EmitterInput], errorDir: String) = {
+    records.groupBy {
+      case Success(byteRecord) =>
+        val strRecord = new String(byteRecord, "UTF-8")
+        Try(parse(strRecord)) match {
+          case TrySuccess(e) =>
+            val json = parse(strRecord)
```
Comment (chuwy) [inline src/main/scala/com.snowplowanalytics.s3/loader/KinesisS3Emitter.scala]: Why do we parse it twice? We already have successful `JValue` in `e`

File: src/main/scala/com.snowplowanalytics.s3/loader/KinesisS3Emitter.scala [hunk 2]
```diff
@@ -108,3 +126,65 @@ class KinesisS3Emitter(
...
+    }
+
+    /** Represents cases where type of row can be determined successfully
+      * e.g. does have proper schema key
+      */
+    case class SelfDescribing(rowType: String) extends RowType {
+      override def getName: String = rowType
+    }
+
+    case object UnexpectedError extends RowType {
```
Comment (chuwy) [inline src/main/scala/com.snowplowanalytics.s3/loader/KinesisS3Emitter.scala]: `UnexpectedError` is a fairly undesriptive name. Does it mean that we waiting for other errors? In our case this error happens only when record is not a valid string (?).

File: src/main/scala/com.snowplowanalytics.s3/loader/KinesisS3Emitter.scala [hunk 3]
```diff
@@ -108,3 +126,65 @@ class KinesisS3Emitter(
...
+    records.groupBy {
+      case Success(byteRecord) =>
+        val strRecord = new String(byteRecord, "UTF-8")
+        Try(parse(strRecord)) match {
+          case TrySuccess(e) =>
+            val json = parse(strRecord)
+            val schemaKey = SchemaKey.extract(json)
+            schemaKey.fold(
+              e => RowType.PartitionError(errorDir),
+              k => RowType.SelfDescribing(s"${k.vendor}.${k.name}")
```
Comment (chuwy) [inline src/main/scala/com.snowplowanalytics.s3/loader/KinesisS3Emitter.scala]: Did we discuess something this output dir? Why there's no model?

File: src/main/scala/com.snowplowanalytics.s3/loader/KinesisS3Emitter.scala (hunk 1 above)
Comment (aldemirenes) [reply src/main/scala/com.snowplowanalytics.s3/loader/KinesisS3Emitter.scala]: Yep, you are right. I think I missed it. I will change it 👍

File: src/main/scala/com.snowplowanalytics.s3/loader/KinesisS3Emitter.scala (hunk 2 above)
Comment (aldemirenes) [reply src/main/scala/com.snowplowanalytics.s3/loader/KinesisS3Emitter.scala]: I think you are right about name being undescriptive. `InvalidRecordString` is much more descriptive I think.

File: src/main/scala/com.snowplowanalytics.s3/loader/KinesisS3Emitter.scala (hunk 3 above)
Comment (aldemirenes) [reply src/main/scala/com.snowplowanalytics.s3/loader/KinesisS3Emitter.scala]: I was planning to ask about it. So, do you think we should add revision-addition too or only model is enough ?

Comment (chuwy) [reply src/main/scala/com.snowplowanalytics.s3/loader/KinesisS3Emitter.scala]: I think it should be `vendor/name/format/model`, but worth to double-check with @yalisassoon

**Comment by chuwy** [comment]
Closing in favor of https://github.com/snowplow/snowplow-s3-loader/pull/150

**Pull request closed without merging**